yarpl/include/yarpl/Refcounted.h

```diff
@@ -146,12 +143,20 @@ class Reference {
 
  private:
   void inc() {
+    static_assert(
```

Comment:
btw. the move of the code here is to be able to declare yarpl::Reference<T> where T is not fully defined. Its very useful feature for header files usages.